File: gcc/testsuite/gfortran.dg/gomp/target-map-iterators-5.f90
! { dg-do compile }
! { dg-options "-fopenmp -fdump-tree-omplower" }

module m
  integer, parameter :: DIM1 = 31
  integer, parameter :: DIM2 = 17
  type :: array_ptr
    integer, pointer :: ptr(:)
 end type
contains
  subroutine f (x, stride)
    type (array_ptr) :: x(:)
    integer :: stride

    !$omp target map(to: x) map(iterator(i=lbound(x, 1):ubound(x, 1):stride), to: x(i)%ptr(:))
    !$omp end target
  end subroutine
end module

! { dg-final { scan-tree-dump-times "D\\\.\[0-9\]+ = __builtin_malloc \\(D\\\.\[0-9\]+\\);" 3 "omplower" } }
! { dg-final { scan-tree-dump-times "__builtin_free \\(omp_iter_data\\\.\[0-9\]+\\);" 3 "omplower" } }

File: gcc/testsuite/c-c++-common/gomp/target-map-iterators-5.c
/* { dg-do compile } */
/* { dg-options "-fopenmp -fdump-tree-omplower" } */

#define DIM2 17

void f (int **x, int lbound, int ubound, int stride)
{
  #pragma omp target map(to:x) map(iterator(i=lbound:ubound:stride), to: x[i][:DIM2])
    ;
}

/* { dg-final { scan-tree-dump-times "_\[0-9\]+ = ubound - lbound;" 2 "omplower" } } */
/* { dg-final { scan-tree-dump-times "D\\\.\[0-9\]+ = __builtin_malloc \\(D\\\.\[0-9\]+\\);" 2 "omplower" } } */
/* { dg-final { scan-tree-dump-times "__builtin_free \\(omp_iter_data\\\.\[0-9\]+\\);" 2 "omplower" } } */
